x/stakeibc/keeper/msg_server_liquid_stake_test.go

```diff
@@ -0,0 +1,215 @@
+		stAtomBalance: sdk.NewInt64Coin("stuatom", 10_000_000),
+	}
+	suite.FundModuleAccount(stakeibc.ModuleName, module.atomBalance)
+	suite.FundModuleAccount(stakeibc.ModuleName, module.stAtomBalance)
+
+	hostZone := stakeibc.HostZone{
+		ChainId:        "GAIA",
+		HostDenom:      "uatom",
+		IBCDenom:       "ibc/uatom",
+		RedemptionRate: sdk.NewDec(1.0),
+	}
+
+	epochTracker := stakeibc.EpochTracker{
+		EpochIdentifier: epochtypes.STRIDE_EPOCH,
+		EpochNumber:     1,
+	}
+
+	initialDepositRecord := recordtypes.DepositRecord{
+		Id:                 1,
+		DepositEpochNumber: 1,
+		HostZoneId:         "GAIA",
+		Amount:             initialDepositAmount,
+	}
+
+	suite.App.StakeibcKeeper.SetHostZone(suite.Ctx, hostZone)
+	suite.App.StakeibcKeeper.SetEpochTracker(suite.Ctx, epochTracker)
+	suite.App.RecordsKeeper.SetDepositRecord(suite.Ctx, initialDepositRecord)
+
+	return TestCase{
+		user:   user,
+		module: module,
+		initialState: State{
+			depositRecordAmount: initialDepositAmount,
+			hostZone:            hostZone,
+		},
+		validMsg: stakeibc.MsgLiquidStake{
+			Creator:   user.acc.String(),
+			HostDenom: "uatom",
+			Amount:    stakeAmount,
+		},
+	}
+}
+
+func (suite *KeeperTestSuite) TestLiquidStakeSuccessful() {
+	tc := suite.SetupLiquidStake()
+	user := tc.user
+	module := tc.module
+	msg := tc.validMsg
+	stakeAmount := sdk.NewInt(msg.Amount)
+
+	_, err := suite.msgServer.LiquidStake(sdk.WrapSDKContext(suite.Ctx), &msg)
+	suite.Require().NoError(err)
+
+	// Confirm balances
+	// User IBC/UATOM balance should have DECREASED by the size of the stake
+	expectedUserAtomBalance := user.atomBalance.SubAmount(stakeAmount)
+	actualUserAtomBalance := suite.App.BankKeeper.GetBalance(suite.Ctx, user.acc, "ibc/uatom")
+	// Module IBC/UATOM balance should have INCREASED by the size of the stake
+	expectedModuleAtomBalance := module.atomBalance.AddAmount(stakeAmount)
+	actualModuleAtomBalance := suite.App.BankKeeper.GetBalance(suite.Ctx, module.acc, "ibc/uatom")
+	// User STUATOM balance should have INCREASED by the size of the stake
+	expectedUserStAtomBalance := user.stAtomBalance.AddAmount(stakeAmount)
+	actualUserStAtomBalance := suite.App.BankKeeper.GetBalance(suite.Ctx, user.acc, "stuatom")
+
+	suite.CompareCoins(expectedUserStAtomBalance, actualUserStAtomBalance, "user stuatom balance")
+	suite.CompareCoins(expectedUserAtomBalance, actualUserAtomBalance, "user ibc/uatom balance")
+	suite.CompareCoins(expectedModuleAtomBalance, actualModuleAtomBalance, "module ibc/uatom balance")
+
+	// Confirm deposit record adjustment
+	records := suite.App.RecordsKeeper.GetAllDepositRecord(suite.Ctx)
+	suite.Require().Len(records, 1, "number of deposit records")
+
+	expectedDepositRecordAmount := tc.initialState.depositRecordAmount + stakeAmount.Int64()
+	actualDepositRecordAmount := records[0].Amount
+	suite.Require().Equal(expectedDepositRecordAmount, actualDepositRecordAmount, "deposit record amount")
```

Comment:
Nit: can we also check the deposit record "status" and "source" are correct?

Reply:
great call, will add that in!

Reply:
Actually, this function shouldn't change those fields, right?

Reply:
I think at this point, `Source` will be nil, and Status will be set to transfer by default, but isn't touched by the liquid stake function